qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/ParameterizedFullClusterRestartTestCase.java

```diff
@@ -43,6 +49,38 @@ public static Iterable<Object[]> parameters() throws Exception {
         return Arrays.stream(FullClusterRestartUpgradeStatus.values()).map(v -> new Object[] { v }).toList();
     }
 
+    @Before
+    public void extractOldIndexVersion() throws Exception {
+        if (upgraded == false) {
+            IndexVersion indexVersion = null;   // these should all be the same version
+
+            Response response = client().performRequest(new Request("GET", "_nodes"));
+            ObjectPath objectPath = ObjectPath.createFromResponse(response);
+            Map<String, Object> nodeMap = objectPath.evaluate("nodes");
+            for (String id : nodeMap.keySet()) {
+                String name = objectPath.evaluate("nodes." + id + ".name");
+
+                Number ix = objectPath.evaluate("nodes." + id + ".index_version");
+                IndexVersion version;
+                if (ix != null) {
+                    version = IndexVersion.fromId(ix.intValue());
+                } else {
+                    String ver = objectPath.evaluate("nodes." + id + ".version");
```

Comment:
this should be OLD_CLUSTER_VERSION right? You could just use that instead of extracting from the node?

Reply:
Ah yes, good spot